// react-bulma-switch/lib/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _label = require('react-bulma-components/lib/components/form/components/label');

var _label2 = _interopRequireDefault(_label);

var _field = require('react-bulma-components/lib/components/form/components/field');

var _field2 = _interopRequireDefault(_field);

var _constants = require('react-bulma-components/lib/constants');

var _constants2 = _interopRequireDefault(_constants);

var _classnames2 = require('classnames');

var _classnames3 = _interopRequireDefault(_classnames2);

require('./index.sass');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Switch = function (_React$PureComponent) {
  _inherits(Switch, _React$PureComponent);

  function Switch(props) {
    _classCallCheck(this, Switch);

    var _this = _possibleConstructorReturn(this, (Switch.__proto__ || Object.getPrototypeOf(Switch)).call(this, props));

    _this.id = props.id || 'bulmaswitch-' + Math.random();
    return _this;
  }

  _createClass(Switch, [{
    key: 'render',
    value: function render() {
      var _classnames;

      var _props = this.props,
          children = _props.children,
          className = _props.className,
          id = _props.id,
          value = _props.value,
          rtl = _props.rtl,
          color = _props.color,
          size = _props.size,
          rounded = _props.rounded,
          outlined = _props.outlined,
          thin = _props.thin,
          props = _objectWithoutProperties(_props, ['children', 'className', 'id', 'value', 'rtl', 'color', 'size', 'rounded', 'outlined', 'thin']);

      return _react2.default.createElement(
        _field2.default,
        { className: className },
        _react2.default.createElement('input', _extends({
          id: this.id,
          type: 'checkbox',
          className: (0, _classnames3.default)('switch', (_classnames = {
            'is-rtl': rtl,
            'is-rounded': rounded,
            'is-outlined': outlined,
            'is-thin': thin
          }, _defineProperty(_classnames, 'is-' + color, color), _defineProperty(_classnames, 'is-' + size, size), _classnames)),
          value: value,
          checked: value
        }, props)),
        _react2.default.createElement(
          _label2.default,
          { htmlFor: this.id },
          children
        )
      );
    }
  }]);

  return Switch;
}(_react2.default.PureComponent);

Switch.propTypes = {
  children: _propTypes2.default.node,
  id: _propTypes2.default.string,
  value: _propTypes2.default.bool,
  rtl: _propTypes2.default.bool,
  rounded: _propTypes2.default.bool,
  outlined: _propTypes2.default.bool,
  thin: _propTypes2.default.bool,
  disabled: _propTypes2.default.bool,
  color: _propTypes2.default.oneOf(Object.values(_constants2.default.COLORS)),
  size: _propTypes2.default.oneOf(['small', 'medium', 'large']),
  className: _propTypes2.default.string
};
Switch.defaultProps = {
  children: null,
  id: null,
  value: false,
  rtl: false,
  rounded: false,
  outlined: false,
  thin: false,
  disabled: false,
  color: null,
  size: null,
  className: null
};
exports.default = Switch;
